### PR TITLE
Update Python versions

### LINKED ISSUE #214: Dropping support for Python 2.7
So given that the ecosystem is [dropping support for 2.7 in earnest](http://python3statement.org), the time has come for us to start planning for Siphon to move on from Legacy Python:

- Core Python developers will [stop support for Python 2.7 January 1, 2020](https://pythonclock.org)
- NumPy feature releases will be [Python 3 only starting January 1, 2019](https://docs.scipy.org/doc/numpy/neps/dropping-python2.7-proposal.html), and support for the last release supporting Python 2 will end January 1, 2020.
- XArray will drop [2.7 January 1, 2019 as well](https://github.com/pydata/xarray/issues/1830)
- Matplotlib's 3.0 release, tentatively Summer 2018, [will be Python 3 only](https://mail.python.org/pipermail/matplotlib-devel/2017-October/000892.html); the current 2.2 release will be the last long term release that supports 2.7.

Given these events, and based on preliminary input from Unidata's Users' committee, I propose that Siphon drop support for Python 2.7 in the Fall of 2019. That implies that the Fall 2019 release of Siphon will be the first that supports only Python 3. The release before that (Spring or Summer 2019) will be the last that supports Python 2.7; given Siphon's relatively nascent status, I also propose that this release NOT have any particular long term support status.

I imagine this will impact some of our users, so I welcome feedback here. Note that we don't have the capacity to maintain all of Siphon's upstream dependencies, so some of this transition is beyond our control.

### PR DESCRIPTION
**Pull request by dopplershift**
#### Description Of Changes
* Finally drop 2.7 (#214)
* Drop Python 3.6
* Update CI a bit

#### Checklist

- [x] Closes #214 
~- [ ] Tests added~
- [x] Fully documented

### CODE REVIEW

File: src/siphon/cdmr/ncstream.py [hunk 1]
```diff
@@ -87,8 +84,7 @@ def read_ncstream_data(fobj):
             magic = read_magic(fobj)
         return data, blocks
     else:
-        raise NotImplementedError("Don't know how to handle data type: {0}".format(
-            data.dataType))
+        raise NotImplementedError(f"Don't know how to handle data type: {data.dataType}")
```
Comment (dcamron) [inline src/siphon/cdmr/ncstream.py:87]:
```suggestion
        raise NotImplementedError(f'Don't know how to handle data type: {data.dataType}')
```
Random double quotes? Didn't check for any more of these.

File: src/siphon/metadata.py [hunk 2]
```diff
@@ -513,7 +511,7 @@ def _get_handler(self, handler_name):
         elif handler_name in self._sts:
             return getattr(self._st, handler_name)
         else:
-            msg = 'cannot find handler for element {}'.format(handler_name)
+            msg = f'cannot find handler for element {handler_name}'
             log.warning(msg)
```
Comment (dcamron) [inline src/siphon/metadata.py:515]: Want to go ahead and flake8-pieify this as a log message (didn't dig for more outside of changes) while you're in here?

File: src/siphon/ncss_dataset.py [hunk 3]
```diff
@@ -201,11 +200,11 @@ def lookup(self, handler_name):
         if handler_name in dir(self):
             return getattr(self, handler_name)
         else:
-            msg = 'cannot find handler for element {}'.format(handler_name)
+            msg = f'cannot find handler for element {handler_name}'
             log.warning(msg)
```
Comment (dcamron) [inline src/siphon/ncss_dataset.py:204]: Same question as `metadata.py:513` above.

**Review by dcamron — APPROVED** [review]
These are all just linting, which we can spare for some other style cleanup if preferred. I'll approve and leave that call up to you

File: src/siphon/cdmr/ncstream.py (hunk 1 above)
Comment (dopplershift) [reply src/siphon/cdmr/ncstream.py:87]: Confused me for a sec just now, but the `'` in "Don't" allows this as double to avoid escaping it.

Comment (dcamron) [reply src/siphon/cdmr/ncstream.py:87]: Haha, totally missed it! Even with Github syntax highlighting pointing it out to me.

File: src/siphon/metadata.py (hunk 2 above)
Comment (dopplershift) [reply src/siphon/metadata.py:515]: Sure. Did yours catch this? Because mine didn't.

File: src/siphon/ncss_dataset.py (hunk 3 above)
Comment (dopplershift) [reply src/siphon/ncss_dataset.py]: Done.

**Comment by dopplershift** [comment]
Thanks for catching those. I just added the logging changes as a separate commit, so re-review should be simple.

File: src/siphon/metadata.py (hunk 2 above)
Comment (dcamron) [reply src/siphon/metadata.py]: I didn't run it myself, I just remembered making that change over on MetPy. I can check and see if it does.

Comment (dcamron) [reply src/siphon/metadata.py]: Update: it doesn't.

**Review by dcamron — APPROVED** [review]

**Comment by dopplershift** [comment]
CodeQL is succeeding, but for some reason is failing to report. 🤷‍♂️

**Pull request merged**